lib/Dialect/FIRRTL/Transforms/LowerOpenAggs.cpp

```diff
@@ -603,18 +631,25 @@ PortMappingInfo Visitor::mapPortType(Type type) {
 
               return BundleType::get(context, hwElements, obTy.isConst());
             })
-        .template Case<OpenVectorType>(
-            [&](OpenVectorType ovTy) -> FIRRTLBaseType {
+            .template Case<OpenVectorType>([&](OpenVectorType ovTy)
+                                               -> FailureOr<FIRRTLBaseType> {
+              uint64_t id = 0;
               FIRRTLBaseType convert;
               // Walk for each index to extract each leaf separately, but expect
               // same hw-only type for all.
               for (auto idx : llvm::seq<size_t>(0U, ovTy.getNumElements())) {
                 auto hwElementType =
                     f(f, ovTy.getElementType(), suffix + "_" + Twine(idx), flip,
-                      fieldID + ovTy.getFieldID(idx));
-                assert((!convert || convert == hwElementType) &&
+                      fieldID + ovTy.getFieldID(idx), newFieldID + id + 1);
+                if (failed(hwElementType))
+                  return failure();
+                assert((!convert || convert == *hwElementType) &&
                        "expected same hw type for all elements");
-                convert = hwElementType;
+                convert = *hwElementType;
+                if (convert)
+                  id += type_cast<hw::FieldIDTypeInterface>(convert)
```

Comment:
Hey I think you can just do `convert->getMaxFieldID()`, `convert` is already `FIRRTLBaseType`.

Reply:
Good call, simplified accordingly, thanks!